tiledb/sm/filter/filter_pipeline.cc

```diff
@@ -599,14 +599,22 @@ Status FilterPipeline::run_reverse_internal(
 
   reader_stats->add_counter("read_unfiltered_byte_num", total_orig_size);
 
-  const Status st = filter_chunks_reverse(
-      *tile, offsets_tile, filtered_chunks, compute_tp, config);
-  if (!st.ok()) {
+  try {
+    const Status st = filter_chunks_reverse(
+        *tile, offsets_tile, filtered_chunks, compute_tp, config);
+    if (!st.ok()) {
```

Comment:
Here we could just throw and the catch block would do the cleanup? Also consider changing the function to throw instead of return status.

Reply:
Thanks for the suggestion! 
Applied.